### DISCUSSION TITLE
2023-06-03 개발일지

### DISCUSSION THREAD

**Discussion by Craftcode17** [Blue-club/pintos1#9] [개발일지]
6/1 : 발제, 채용설명회, 소독 등으로 협업시간 고정하기 어려움. 자율적으로 학습
6/2 : Project2 관련 개념 스스로 톺아보기, 외부특강 대비 
6/3 : (공통) 14:00~ 16:30 외부특강 - 권영진 교수님, OS 특강

**Comment by Riudiu** [comment]
오늘 한 것(6/3)
- 알고리즘 문제풀이 - LIS 3,4 
- KAIST 권영진 교수님 OS특강 1차
- [비전공자를 위한 컴퓨터 구조 및 OS 개론 강의](https://www.inflearn.com/course/%EB%B9%84%EC%A0%84%EA%B3%B5%EC%9E%90-%EC%BB%B4%ED%93%A8%ED%84%B0%EA%B5%AC%EC%A1%B0-os%EA%B0%9C%EB%A1%A0)

학습 내용
- 기본적인 컴퓨터구조와 운영체제 지식 복습 및 전반적으로 훑어보기

**Comment by Craftcode17** [comment]
오늘 한 것(6/3 토)
- OS특강 수강, 내용 복습
- 작업순서 : 
    - 수정해야 할 부분 정리
    - 수정할 부분 설계(수도코드 작성)
    - 관련 테스트 수행, 테스트 통과 후 다음 내용 진행.
    - 사이사이에 필요한 부분 학습, 내용 공유
    -> 미리 공부하면 기억에 남지 않음. 
    
- Project2 깃북 정독,번역, 해야할 일 정리(진행중)
    - Argument Passing 구현 : process_exec() 함수 수정
    공백 기준으로 명령어에서 파일명과 인수를 구분지어 인식하도록 수정하자
    명령줄 길이에 합리적인 제한을 주어도 OK
    `strtok_r()`, `lib/string.c` 의 상세주석 참고할 것 (String을 다루는 부분)
    - User Memory Access 구현 : 포인터에서 데이터를 읽을 때 해당 기능을 통해 프록시.
    잘못된 포인터 등을 받으면 프로세스 terminated 방식으로 케이스 핸들링

    - System Call 시스템 호출 : `userprot/syscall.c` 수정, 아래의 함수목록(프로토타입) 수정 구현([깃북 상세](https://casys-kaist.github.io/pintos-kaist/project2/system_call.html))
        - `void halt (void);` : 
        `power_off()` 호출하여 Pintos를 종료합니다. (`src/include/threads/init.h` 에서 선언됨). 
        가능한 교착 상태 등에 대한 일부 정보를 잃기 때문에 거의 사용하지 않아야 합니다.
        - `void exit (int status);`
        현재 사용자 프로그램을 종료하고 `status`커널로 돌아갑니다. 
        프로세스의 부모가 `wait`이를 위한 경우(아래 참조) 반환될 상태입니다. 
        일반적으로 `status` 가 `0`인 경우 성공을 나타내고, 0이 아닌 값은 오류를 나타냅니다.
        - `pid_t fork (const char *thread_name);`
        - `int exec (const char *cmd_line);`
        - `int wait (pid_t pid);`
        - `bool create (const char *file, unsigned initial_size);`
        - `bool remove (const char *file);`
        - `int open (const char *file);`
        - `int filesize (int fd);`
        - `int read (int fd, void *buffer, unsigned size);`
        - `int write (int fd, const void *buffer, unsigned size);`
        - `void seek (int fd, unsigned position);`
        - `unsigned tell (int fd);`
        - `void close (int fd);`
